src/Control/Effect/Resumable.hs

```diff
@@ -98,22 +103,22 @@ instance (Carrier sig m, Effect sig) => Carrier (Resumable err :+: sig) (Resumab
 --
 --   prop> run (runResumableWith (\ (Err b) -> pure (1 + b)) (pure a)) == a
 --   prop> run (runResumableWith (\ (Err b) -> pure (1 + b)) (throwResumable (Err a))) == 1 + a
-runResumableWith :: (Carrier sig m, Monad m)
-                 => (forall x . err x -> m x)
-                 -> Eff (ResumableWithC err m) a
+runResumableWith :: (forall x . err x -> m x)
+                 -> ResumableWithC err m a
                  -> m a
-runResumableWith with = runResumableWithC with . interpret
+runResumableWith with = runReader (Handler with) . runResumableWithC
 
-newtype ResumableWithC err m a = ResumableWithC ((forall x . err x -> m x) -> m a)
+newtype ResumableWithC err m a = ResumableWithC { runResumableWithC :: ReaderC (Handler err m) m a }
+  deriving (Alternative, Applicative, Functor, Monad, MonadFail, MonadIO, MonadPlus)
 
-runResumableWithC :: (forall x . err x -> m x) -> ResumableWithC err m a -> m a
-runResumableWithC f (ResumableWithC m) = m f
+instance MonadTrans (ResumableWithC err) where
+  lift m = ResumableWithC (lift m)
 
-instance (Carrier sig m, Monad m) => Carrier (Resumable err :+: sig) (ResumableWithC err m) where
-  ret a = ResumableWithC (const (ret a))
-  eff op = ResumableWithC (\ handler -> handleSum
-    (eff . handlePure (runResumableWithC handler))
-    (\ (Resumable err k) -> handler err >>= runResumableWithC handler . k) op)
+newtype Handler err m = Handler { runHandler :: forall x . err x -> m x }
```

Comment:
This helper `newtype` enables us to use `ReaderC` in the definition of `ResumableWithC`.